src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -2117,7 +2132,20 @@ public void CreateModel(XmlElement modelData)
                 var connector = NodeGraph.LoadConnectorFromXml(modelData,
                     Nodes.ToDictionary(node => node.GUID));
 
-                OnConnectorAdded(connector); // Update view-model and view.
+                // It is possible that in some cases connector can't be created,
+                // for example, connector connects to a custom node instance
+                // whose input ports have been changed, so connector can't find
+                // its end port owner.
+                if (connector == null)
+                {
+                    XmlAttribute guidAttribute = modelData.Attributes["guid"];
+                    if (null != guidAttribute)
+                        undoRecorder.RecordOffTrackModel(Guid.Parse(guidAttribute.Value)); 
```

Comment:
I'd prefer to do this instead of silently make it pass if `guidAttribute` is `null`:

```csharp
if (guidAttribute == null)
{
    throw new InvalidOperationException("'guid' field missing from recorded model");
}
```

I have been advised not to do `null != guidAttribute` and do `guidAttribute != null` before, so I'll pass on this advice. Also single-line return statement should have curly brackets around it, according to our coding standard ;)